diff --git a/src/SamlService.php b/src/SamlService.php
index 37288fe..2205e76 100644
--- a/src/SamlService.php
+++ b/src/SamlService.php
@@ -12,6 +12,8 @@ use Drupal\Core\Session\AnonymousUserSession;
 use Drupal\Core\StringTranslation\StringTranslationTrait;
 use Drupal\Core\StringTranslation\TranslationInterface;
 use Drupal\Core\Url;
+use Drupal\Core\Utility\Token;
+use Drupal\Core\Render\BubbleableMetadata;
 use Drupal\externalauth\Authmap;
 use Drupal\externalauth\ExternalAuth;
 use Drupal\key\KeyRepositoryInterface;
@@ -1079,10 +1081,13 @@ class SamlService {
    *   The library configuration array.
    */
   protected static function reformatConfig(ImmutableConfig $config, $base_url = '', $purpose = '', ?KeyRepositoryInterface $key_repository = NULL) {
+    $token_service = \Drupal::token();
+    $bubbleable_metadata = new BubbleableMetadata();
+
     $library_config = [
       'debug' => (bool) $config->get('debug_phpsaml'),
       'sp' => [
-        'entityId' => $config->get('sp_entity_id'),
+        'entityId' => $token_service->replace($config->get('sp_entity_id'), [], [], $bubbleable_metadata),
         'assertionConsumerService' => [
           // See ExecuteInRenderContextTrait if curious why the long chained
           // call is necessary.
